sourcecode/apis/contentauthor/app/Console/Libraries/CliH5pAjax.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Console\Libraries;
+
+use App;
+use App\Libraries\ContentAuthorStorage;
+use App\Libraries\H5P\EditorAjax;
+use App\Libraries\H5P\EditorStorage;
+use H5PCore;
+use H5peditor;
+use H5PEditorEndpoints;
+use H5PFileStorage;
+use RuntimeException;
+
+class CliH5pAjax
+{
+    public static function installLibrary(string $machineName): string
+    {
+        if (!App::runningInConsole()) {
+            throw new RuntimeException('Interface not supported');
+        }
+
+        // H5P Editor preforms permission checks when installing/updating libraries, so we override
+        // the permission functions in a CLI version of the Framework
+        $cliFramework = app(CliH5pFramework::class);
+
+        $fileStorage = app(H5PFileStorage::class);
+        $contentAuthorStorage = app(ContentAuthorStorage::class);
+        $core = new H5PCore($cliFramework, $fileStorage, $contentAuthorStorage->getAssetsBaseUrl());
```

Comment:
There is an open PR to remove `getAssetsBaseUrl()`, but only the `hubbub` branch. This isn't a problem if the service container is changed so `app(H5PFrameworkInterface::class)` returns the appropriate instance instead, however. For instance:

```php
$this->app->singleton(H5PFrameworkInterface::class, function () {
    $framework = new Framework(/* ... */);
    if ($this->app->runningInConsole()) {
        return new CliH5pFramework(/* ... */);
    }

    return new Framework(/* ... */);
});
```

This removes the need to manually instantiate the entire dependency tree for the `$ajax` object, which creates a maintenance burden.